h2o-core/src/main/java/water/api/RequestServer.java

```diff
@@ -218,31 +218,48 @@ public void doGeneric(String method, HttpServletRequest request, HttpServletResp
         headers.put(key, value);
       }
 
+      final String contentType = request.getContentType();
       Properties parms = new Properties();
-      Map<String, String[]> parameterMap;
-      parameterMap = request.getParameterMap();
-      for (Map.Entry<String, String[]> entry : parameterMap.entrySet()) {
-        String key = entry.getKey();
-        String[] values = entry.getValue();
-
-        if (values.length == 1) {
-          parms.put(key, values[0]);
-        } else if (values.length > 1) {
-          StringBuilder sb = new StringBuilder();
-          sb.append("[");
-          boolean first = true;
-          for (String value : values) {
-            if (!first) sb.append(",");
-            sb.append("\"").append(value).append("\"");
-            first = false;
+      String postBody = null;
+
+      if ("application/json".equals(contentType)) {
+        StringBuffer jb = new StringBuffer();
+        String line = null;
+        try {
+          BufferedReader reader = request.getReader();
+          while ((line = reader.readLine()) != null)
+            jb.append(line);
+        } catch (Exception e) {
+          throw new H2OIllegalArgumentException("Exception reading POST body JSON for URL: " + uri);
+        }
+        postBody = jb.toString();
+      } else {
+        // application/x-www-form-urlencoded
+        Map<String, String[]> parameterMap;
+        parameterMap = request.getParameterMap();
+        for (Map.Entry<String, String[]> entry : parameterMap.entrySet()) {
+          String key = entry.getKey();
+          String[] values = entry.getValue();
+
+          if (values.length == 1) {
+            parms.put(key, values[0]);
+          } else if (values.length > 1) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("[");
+            boolean first = true;
+            for (String value : values) {
+              if (!first) sb.append(",");
+              sb.append("\"").append(value).append("\"");
+              first = false;
+            }
+            sb.append("]");
+            parms.put(key, sb.toString());
```

Comment:
I didn't change this code; it's just an indent change because it's now inside an if.  The purpose of the code is to properly handle arrays passed as application/x-www-form-urlencoded, where the values in the array actually get sent as multiple individual parameters something like this:

?my_array="first value"&my_array="second value"

The code glues these together into a single string like my_array=["first value", "second value"] to be handled by the value parser inside Schema.

So: no, that won't work.